corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -140,16 +138,26 @@ def _update_case_search_labels(self, rows):
                     self.msgs.append((messages.error, message))
                     continue
                 self._update_translation(display_row, prop.label)
-            for hint_row, prop in itertools.chain(zip(hints, properties)):
-                if hint_row.get('case_property') != prop.name:
-                    message = _('A hint row for menu {index} has an unexpected case search property "{field}". '
-                                'Case properties must appear in the same order as they do in the bulk '
-                                'app translation download. No translations updated for this row.').format(
-                                    index=self.module.id + 1,
-                                    field=hint_row.get('case_property', ""))
-                    self.msgs.append((messages.error, message))
-                    continue
-                self._update_translation(hint_row, prop.hint)
+            if len(hints) == len(properties):
+                for hint_row, prop in itertools.chain(zip(hints, properties)):
+                    if hint_row.get('case_property') != prop.name:
+                        message = _('A hint row for menu {index} has an unexpected case search property "{field}".'
+                                    ' Case properties must appear in the same order as they do in the bulk '
+                                    'app translation download. No translations updated for this row.').format(
+                                        index=self.module.id + 1,
+                                        field=hint_row.get('case_property', ""))
+                        self.msgs.append((messages.error, message))
+                        continue
+                    self._update_translation(hint_row, prop.hint)
+            elif hints and len(hints) != len(properties):
```

Comment:
nit: couldn't this just be `else`? `len(hints) != len(properties)` is given since `len(hints) == len(properties)` is false and hints would just be an empty list which wouldn't be a problem.

Reply:
I didn't really want the hint update block to run if there were 0 hints (which I think is pretty common) and the `len(hints) != len(properties)` was really just to make it clear that it should update hints selectively when there are few, but not for all displays

Reply:
the `for` loop wouldn't iterate if there were 0 hints `[]`. Also see your point about being explicit. I'm okay with keeping it the way you have it